universal-login-relayer/src/integration/ethereum/validators/MessageExecutionValidator.ts

```diff
@@ -4,14 +4,15 @@ import {ComposeValidator} from '../../../core/services/validators/ComposeValidat
 import CorrectProxyValidator from './CorrectProxyValidator';
 import CorrectMasterValidator from './CorrectMasterValidator';
 import EstimateGasValidator from './EstimateGasValidator';
+import {WalletContractService} from '../WalletContractService';
 
 export class MessageExecutionValidator extends ComposeValidator {
-  constructor(wallet: Wallet, contractWhiteList: ContractWhiteList) {
+  constructor(wallet: Wallet, contractWhiteList: ContractWhiteList, walletContractService: WalletContractService) {
```

Comment:
You can take provider from WalletContractService (make it readonly instead of private)